docs/modules/ROOT/pages/cop_api_v1_changelog.adoc

```diff
@@ -0,0 +1,258 @@
+= Cop API v0 to v1 API changes
+:doctype: book
+
+== Upgrade guide
+
+Your custom cops should continue to work in v1.
+
+Nevertheless it is suggested that you tweak them to use the v1 API by following the following steps:
+
+1) Your class should inherit from `RuboCop::Cop::Base` instead of `RuboCop::Cop::Cop`.
+
+2) Locate your calls to `add_offense` and make sure that you pass as the first argument either an `AST::Node`, a `::Parser::Source::Comment` or a `::Parser::Source::Range`, and no `location:` named parameter.
+
+[discrete]
+==== Example:
+
+[source,ruby]
+----
+# Before
+class MySillyCop < Cop
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node, location: :selector, message: "Be positive")
+    end
+  end
+end
+
+# After
+class MySillyCop < Base
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node.loc.selector, message: "Be positive")
+    end
+  end
+end
+----
+
+=== If your class supports autocorrection
+
+Your class must `extend Autocorrector`.
+
+The `corrector` is now yielded from `add_offense`. Move the code of your method `autocorrect` in that block and do not wrap your correction in a lambda. `Corrector` are more powerful and can now be `merge`d.
+
+==== Example:
+
+[source,ruby]
+----
+# Before
+class MySillyCorrectingCop < Cop
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node, location: :selector, message: 'Be positive')
+    end
+  end
+
+  def autocorrect(node)
+    lambda do |corrector|
+      corrector.replace(node.loc.selector, '+')
+    end
+  end
+end
+
+# After
+class MySillyCorrectingCop < Base
+  extend Autocorrector
+
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node.loc.selector, message: 'Be positive') do |corrector|
+        corrector.replace(node.loc.selector, '+')
+      end
+    end
+  end
+end
+----
+
+=== If your class uses instance variables
+
+Do not use RuboCop's internal instance variables. If you used `@processed_source`, use `processed_source`. If you have a need to access an instance variable, open an issue with your use case.
+
+Cop instances will be called on different `processed_source` if the class supports it. In all cases it is recommended to to invalidate your cache data during the `on_new_investigation` callback:
+
+[source,ruby]
+----
+# Before
+class MyCachingCop < Cop
+  def on_send(node)
+    if my_cached_data[node]
+      #...
+    end
+  end
+
+  def my_cached_data
+    @data ||= processed_source.comments.map { # ... }
+  end
+end
+
+# After
+class MyCachingCop < Base
+  # ... same as above, plus...
+  def on_new_investigation
+    @data = nil
+    super  # Be nice and call super for callback
+  end
+end
+----
+
+=== Other API changes
+
+If your cop uses `investigate`, `investigate_post_walk`, `join_force?`, or internal classes like `Corrector`, `Commissioner`, `Team`, these have changed. See the Detailed API Changes.
```

Comment:
You can add an anchor to "Detailed API Changes" here.

Reply:
Absolutely.